src/animation_player.rs

```diff
@@ -98,44 +116,80 @@ pub fn match_bones(
-                                offset_z += frame[&joint.data().channels()[2]];
+                                current_offset_x += current_frame[&joint.data().channels()[0]];
+                                current_offset_y += current_frame[&joint.data().channels()[1]];
+                                current_offset_z += current_frame[&joint.data().channels()[2]];
+
+                                next_offset_x += next_frame[&joint.data().channels()[0]];
+                                next_offset_y += next_frame[&joint.data().channels()[1]];
+                                next_offset_z += next_frame[&joint.data().channels()[2]];
+
+                                current_rotation0 = current_frame[&joint.data().channels()[3]];
+                                current_rotation1 = current_frame[&joint.data().channels()[4]];
+                                current_rotation2 = current_frame[&joint.data().channels()[5]];
 
-                                rotation0 = frame[&joint.data().channels()[3]];
-                                rotation1 = frame[&joint.data().channels()[4]];
-                                rotation2 = frame[&joint.data().channels()[5]];
+                                next_rotation0 = next_frame[&joint.data().channels()[3]];
+                                next_rotation1 = next_frame[&joint.data().channels()[4]];
+                                next_rotation2 = next_frame[&joint.data().channels()[5]];
                             }
 
-                            let rotation = Quat::from_euler(
+                            let current_rotation = Quat::from_euler(
                                 EulerRot::ZYX,
-                                rotation0.to_radians(),
-                                rotation1.to_radians(),
-                                rotation2.to_radians(),
+                                current_rotation0.to_radians(),
+                                current_rotation1.to_radians(),
+                                current_rotation2.to_radians(),
                             );
 
+                            let next_rotation = Quat::from_euler(
+                                EulerRot::ZYX,
+                                next_rotation0.to_radians(),
+                                next_rotation1.to_radians(),
+                                next_rotation2.to_radians(),
+                            );
+
+                            let current_position =
+                                Vec3::new(current_offset_x, current_offset_y, current_offset_z);
+
+                            let next_position =
+                                Vec3::new(next_offset_x, next_offset_y, next_offset_z);
+
+                            let interpolated_position =
+                                current_position.lerp(next_position, interpolation_factor);
+
+                            let interpolated_rotation =
+                                current_rotation.slerp(next_rotation, interpolation_factor);
```

Comment:
just a preference, I like using:
```rust
Vec3::lerp(current_position, next_position, interpolation_factor);
```
instead of:
```rust
current_position.lerp(next_position, interpolation_factor);
```

Same goes to `Quat`.